src/rpc/blockchain.cpp

```diff
@@ -706,7 +711,7 @@ UniValue getblock(const UniValue& params, bool fHelp)
     // If height is supplied, find the hash
     if (strHash.size() < (2 * sizeof(uint256))) {
         // std::stoi allows characters, whereas we want to be strict
-        regex r("[[:digit:]]+");
+        regex r("-?[[:digit:]]+");
```

Comment:
This allows `-0` (interpreted as `0`, i.e. the genesis block). Is that intended?

Reply:
Its unintended. Thanks.

I tried to find something better online but no luck so built this one:

`(?:(-?)[1-9][0-9]*|[0-9]+)`

Where:

- `?:` - if
- `(-?)` - Condition, string starting with `-` character.
- `[1-9]` - Then next character must be a digit between 1 and 9.
- `[0-9]*` - Can be more digits from 0 to 9.
- `|` - Else
- `[0-9]+` One or more 0 to 9 digits.

https://github.com/zcash/zcash/pull/4280/commits/3fb8af512c464140d6ca67c814313bd72bb17780